### ISSUE TITLE
Text selection for TextBlock And AutoSize for RichTextBox 

### ISSUE THREAD

**Issue by Red6789** [dotnet/wpf#7940]
When you use WPF to develop chat applications, you will find that you need to use the TextBlock control to display rich text and allow the user to select and copy text is a very common need, if you use TextBox to implement it lacks TextBlock's Inlines property, if you use RichTextBox to implement it lacks TextBlock's automatic width and height properties, I hope to add the function of the selection effect to TextBlock. RichTextBox adds automatic height and width

**Comment by Patch4747** [comment]
Agreed, `TextBlock` needs an easy way to enable text selection. Current workarounds either lose some `TextBlock` behavior (as mentioned by @Red6789) or they just mimic text selection instead of the native text selection feature.

It's worth noting that UWP has [TextBlock.IsTextSelectionEnabled](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.textblock.istextselectionenabled) property and WPF should too!

**Comment by lindexi** [comment]
@Patch4747 How about the ReadOnly TextBox ?

**Comment by miloush** [comment]
@lindexi it said in the issue it needs rich text

**Comment by dotMorten** [comment]
Text selection is duplicate of https://github.com/dotnet/wpf/issues/307
